### PR TITLE
Do deep sort instead of shallow sort. Allow sort.js to be run without…

### PR DESCRIPTION
**Pull request by michal-huzevka**
… any arguments

### CODE REVIEW

File: sort.js [hunk 1]
```diff
@@ -8,19 +9,48 @@ node sort.js {arguments}
...
+    let result = {};
+    
+    sortedKeys.forEach((key) => {
+        const val = obj[key];
+
+        // Do not sort arrays, as the order of elements is important.
+        // E.g. reordering questions-api.json's characterMapCharacters would
+        // cause the special characters to appear in a different order in
+        // the product.
+        if (val && !Array.isArray(val) && typeof val === 'object') {
```
Comment (shtrom) [inline sort.js]: Do you need the `!Array.isArray(val)` here? Or are arrays `object`s in JS? If so, I'd swap the `Array.isArray` and the `typeof`

**Review by shtrom — APPROVED** [review]
Looks good to me, but I would prefer if the data changes were in a separate commit to the code change.

File: sort.js (hunk 1 above)
Comment (SalimBensiali) [reply sort.js]: @shtrom welcome to `typeof` confusion land :)
image

http://web.mit.edu/jwalden/www/isArray.html

Comment (michal-huzevka) [reply sort.js]: Yep, arrays are considered `object`s in JS. I'll swap the two then.

**Pull request merged**